### ISSUE TITLE
method comparision in GP

### ISSUE THREAD

**Issue by yorkerlin** [shogun-toolbox/shogun#3135]
@karlnapf 
reference:
http://scikit-learn.org/dev/auto_examples/classification/plot_classifier_comparison.html

**Comment by yorkerlin** [comment]
For example, we can use different kernels and inference methods.

**Comment by yorkerlin** [comment]
@karlnapf 
I think it will be great if we can automatically include examples/cookbooks in Shoung's API documentation.

See notebook/examples in sklearn's GPR
http://scikit-learn.org/dev/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html#sklearn.gaussian_process.GaussianProcessRegressor

**Comment by karlnapf** [comment]
Integrating this is exactly the point of the cookbook. So yes, once GSoC is over, we have many more than now and we can do this.

BTW there already is a notebook that compares the classifiers #3019

**Comment by yorkerlin** [comment]
@karlnapf 
this issue is GP only :).
@karlnapf 
For classification in GP, there are many components to choose. eg, kernels, likelihood (eg, logistic, probit), inference methods.

**Comment by karlnapf** [comment]
Why would that be different from what we have?

**Comment by yorkerlin** [comment]
I think  decision boundary may be different even when training accuracy is the same.

**Comment by yorkerlin** [comment]
@karlnapf 
for example,  Decision boundary in linear SVM is different from the one in RBF SVM.

**Comment by karlnapf** [comment]
Maybe we are misunderstanding each other: Such plots are already in that notebook I referenced above. Kernel SVM vs linear SVM vs GP. I don't see a reason to do another comparison for GPs. Why do you wanna do that?

**Comment by yorkerlin** [comment]
For example, `GP using RBF kernel`  vs `GP using linear kernel`.

**Comment by yorkerlin** [comment]
BTW, only GP with `RBF` kernel is used in that plots.

**Comment by karlnapf** [comment]
Well I think this is independent of GPs.
What about a cookbook on covariance functions and kernels in Shogun?

**Comment by yorkerlin** [comment]
https://github.com/yorkerlin/plots-for-GP/blob/master/Untitled1.ipynb

**Comment by karlnapf** [comment]
Do you maybe want to put that in a new (polished) version and put in a separate notebook?

**Comment by yorkerlin** [comment]
Add a notebook/cookbook about http://people.seas.harvard.edu/~dduvenaud/cookbook/ using the Shogun's GP.
